controllers/territory_controller/tests.py

```diff
@@ -184,3 +187,91 @@ def test_BE_links(self) -> None:
             {StationCode.EY, StationCode.VB, StationCode.SZ, StationCode.PO},
             'Territory BE has incorrect territory links',
         )
+
+
+class TestTerritoryLockout(unittest.TestCase):
+    "Test that individual territories become 'locked' when claimed a set number of times"
+
+    _zone_0_territories = {StationCode.PN, StationCode.EY}
+    _zone_1_territories = {StationCode.YL, StationCode.PO}
+
+    def load_territory_owners(self, claim_log: ClaimLog) -> None:
+        for territory in self._zone_0_territories:
+            claim_log._station_statuses[territory] = Claimant.ZONE_0
+
+        for territory in self._zone_1_territories:
+            claim_log._station_statuses[territory] = Claimant.ZONE_1
+
+    @patch('controller.Supervisor.getFromDef')
+    @patch('territory_controller.get_robot_device')
+    @patch('territory_controller.TerritoryController.set_score_display')
+    def setUp(
+        self,
+        mock_score_display: Mock,
+        mock_get_robot: Mock,
+        mock_getFromDef: Mock,
+    ) -> None:
+        super().setUp()
+        claim_log = ClaimLog(record_arena_actions=False)
+        self.load_territory_owners(claim_log)
+        self.attached_territories = AttachedTerritories(claim_log)
+        self.territory_controller = TerritoryController(
+            claim_log,
+            self.attached_territories,
+        )
+
+    @patch('controller.Supervisor.getFromDef')
+    def test_territory_lockout(self, mock_getFromDef: Mock) -> None:
+        """
+        Test a territory is locked after the correct number of claims and
+        disconnected territories aren't also locked
+        """
+        for i in range(LOCKED_OUT_AFTER_CLAIM):
+            # lock status is tested before capturing since the final
+            # iteration will lock the territory
+            for station in [StationCode.BE, StationCode.SZ]:  # assert BE, SZ unlocked
+                self.assertFalse(
+                    self.territory_controller._claim_log.is_locked(station),
+                    f"Territory {station.value} locked early on claim {i}",
+                )
+
+            # capture BE, SZ by alternating sides
+            if i % 2 == 0:
+                claimant = Claimant.ZONE_0
+            else:
+                claimant = Claimant.ZONE_1
+            for station in [StationCode.BE, StationCode.SZ]:
+                # when BE becomes locked SZ will be uncapturable
+                # which will prevent it from also locking
+                self.territory_controller.claim_territory(station, claimant, 0)
```

Comment:
Sorry, I should have been clearer about this -- my concern here is that using things like modulo to determine behaviours creates complex implicit couplings between the conditional block and the loop iterations. This is generally a bad idea in live code and highly undesirable in tests. In tests, I strongly discourage the use of conditionals and all but the simplest of loops.

If this does need to be loops then something like this is probably clearer:
```python
        for claimant, station in (
            (Claimant.ZONE_0, StationCode.BE),
            (Claimant.ZONE_1, StationCode.SZ),
        ):
            with self.subTest(claimant=claimant, station=station):
                for i in range(LOCKED_OUT_AFTER_CLAIM):
                    self.assertFalse(
                        self.territory_controller._claim_log.is_locked(station),
                        f"Territory {station.value} locked early on claim {i}",
                    )

                self.territory_controller.claim_territory(station, claimant, 0)
```
Though I'd be tempted to just inline the calls.

Reply:
https://github.com/srobo/competition-simulator/pull/320/commits/0c3292a16be1d0eeb2de0b44a9c353460a796bce